File: python_backend/game_logic/ranges/jumps/get_jumps.py
import copy
from game_logic.printers.print_board import print_board
from game_logic.ranges.jumps.get_init_jumps import get_init_jumps
from game_logic.ranges.jumps.filter_init_jumps import filter_init_jumps
from game_logic.test_objects.sample_board_dicts import sample_board_dicts


def get_jumps(offsets, board, sqr, color):
    """jumps are square that can be reached through a piece jumping"""
    jumps = get_init_jumps(offsets, board, sqr, color)
    jumps = filter_init_jumps(jumps, board, color)
    return jumps


if __name__ == "__main__":
    offsets = [
        [1, 1],
        [1, 2],
        [1, 3],
        [-1, 1],
        [-1, 2],
        [-1, 3],
        [1, -1],
        [1, -2],
        [1, -3],
        [-1, -1],
        [-1, -2],
        [-1, -3]]

    board = sample_board_dicts['castle_test1']

    jumps = get_jumps(copy.deepcopy(offsets), board, (1, 3), 'W')
    print_board(board, highlights=jumps)

    jumps = get_jumps(copy.deepcopy(offsets), board, (5, 5), 'B')
    print_board(board, highlights=jumps)

File: app/src/game_logic/ranges/jumps/getJumps.ts
import { getInitJumps } from "./getInitJumps";
import { filterInitJumps } from "./filterInitJumps";
import Color from "../../types/Color";
import XY from "../../types/XY";
import Board from "../../types/Board";

export function getJumps(offsets: XY [], board: Board, sqr: XY, color: Color) {
    /*jumps are square that can be reached through a piece jumping**/
    let jumps = getInitJumps(offsets, sqr);
    jumps = filterInitJumps(jumps, board, color);
    return jumps;
}
